src/puppetlabs/puppetdb/utils.clj

```diff
@@ -310,6 +312,12 @@
                 #(update % 0 underscores->dashes)
                 m))
 
+(defn synthesize-body-str
+  [{:strs [command version certname payload]}]
+  (json/generate-string
+    {:command command :version (edn/read-string version)
```

Comment:
Here as well, probably don't want edn. Maybe move the coersion fn into this file and use it?